### DISCUSSION TITLE
how to properly package whl for mac?

### DISCUSSION THREAD

**Discussion by shi-yan** [PyO3/maturin#807] [Q&A] [answered]
Hello,

I have seen instructions on how to build whl for linux using the manylinux docker image, so that we can be sure the module will be loadable on different distros. 

I don't seem to find an equivalent instruction for mac? is it because mac whl is by default universal? or is it because we don't have container images for mac?

Thanks

**Comment by messense** [comment, marked as answer]
Unfortunately there isn't a standard like manylinux for packaging macOS wheels, but in practice [delocate](https://github.com/matthew-brett/delocate) can be used.

If you're building on CI, consider using [cibuildwheel](https://github.com/pypa/cibuildwheel) to build macOS wheels if it has dylib dependencies, cibuildwheel will run delocate for you.

**Reply by shi-yan** [reply]
Thank you very much! 

I just found out cibuildwheel too, but I wan't able to get it to work. All the docker images it uses don't have rust/cargo. I tried to install it before building, but I couldn't config the environment variable properly, it couldn't find the installed rust/cargo.

**Reply by shi-yan** [reply]
thank you. I just also find cibuildwheel, but I couldn't get it to work. 

all the docker images it uses don't have rust/cargo installed. I tried installing them before building, but I couldn't configure the environment variable $PATH properly. it can't find the installed cargo.

**Reply by messense** [reply]
Take a look at this example: https://github.com/Daggy1234/polaroid/blob/main/.github/workflows/publish.yml
